verification/src/compiler/compilers.rs

```diff
@@ -31,19 +32,31 @@ impl Compilers {
             fetcher,
         }
     }
-
+    #[instrument(name = "download_and_compile", skip(self, input), level = "debug")]
     pub async fn compile(
         &self,
         compiler_version: &compiler::Version,
         input: &CompilerInput,
     ) -> Result<CompilerOutput, Error> {
-        let solc_path = match self.cache.get(&*self.fetcher, compiler_version).await {
+        let solc_path_result = {
+            let span = tracing::debug_span!("get solc_path from cache");
+            self.cache
+                .get(self.fetcher.as_ref(), compiler_version)
+                .instrument(span)
```

Comment:
For the most cases imo it's better to put the span inside the function, if possible
Like here, just put it inside cache

Reply:
this span just doesn't attach to current trace without it. here is more about it:
https://docs.rs/tracing/latest/tracing/struct.Span.html#in-asynchronous-code

Reply:
I'm talking about just writing the macro for this function
![image](https://user-images.githubusercontent.com/14099557/184174197-e74d03ec-8dee-416a-9e36-3b75adcc6cee.png)